src/controller/AccountController.js

```diff
@@ -0,0 +1,51 @@
+import { Account } from '../model/Account.js';
+import { AccountCollection } from '../collection/AccountCollection.js';
+import { AccountView } from '../view/AccountView.js';
+
+export class AccountController {
+  constructor() {
+    this.AccountCollection = new AccountCollection();
+    this.accountView = new AccountView();
+  }
+
+  getAccountCount() {
+    return this.AccountCollection.renderLengthAccounts();
+  }
+
+  createAccount(name, initialBalance) {
+    try {
+      const account = new Account(name, initialBalance);
+      this.AccountCollection.create(account)
+      return this.accountView.renderAccountDetails(account);
+    } catch (error) {
+      throw error;
+    }
+  }
+
+  editAccount(accountId, name) {
+    try {
+      const account = this.AccountCollection.update(accountId, name, 'name');
+      return this.accountView.renderAccountDetails(account);
+    } catch (error) {
+      throw error;
+    }
+  }
+
+  depositAccount(accountId, deposit) {
+    try {
+      const account = this.AccountCollection.update(accountId, deposit, 'deposit');
+      return this.accountView.renderAccountDetails(account);
+    } catch (error) {
+      throw error;
+    }
+  }
+
+  getAccountDetails(accountId) {
+    try {
+      const account = this.AccountCollection.findById(accountId);
+      return this.accountView.renderAccountDetails(account);
+    } catch (error) {
+      throw error;
+    }

```

Comment:
if you're catching an error just to throw it again, you're not really handling it. always ensure errors are handled properly. In some edge cases, you might not know exactly what to do, but you should still try to handle them gracefully."

```suggestion
    } catch (error) {
      // Check if the error type is recoverable
      // and handle it gracefully, otherwise rethrow or log it
    }
```